src/scalar/text.ts

```diff
@@ -0,0 +1,47 @@
+import { Utf8 as ArrowString } from '@apache-arrow/esnext-esm';
+
+import { isInvalid, NULL_VALUE } from './util.js';
+
+export class Text {
```

Comment:
I added a new interface that should help implement scalars, can you try adding it, see:
https://github.com/cloudquery/plugin-sdk-javascript/blob/c7b301dfd20d4ce6dd1268090eedf0a4718fc2f9/src/scalar/bool.ts#L7